File: backend/api/src/altasai/evaluation/evaluationRunner.ts
import { evaluateEntities } from './evaluateEntities';
import { evaluateIntent } from './evaluateIntent';
import { evaluateRecommendations } from './evaluateRecommendations';
import { evaluateReflection } from './evaluateReflection';
import { evaluateReports } from './evaluateReports';
import { evaluateRiskModels } from './evaluateRiskModels';
import { evaluateSafety } from './evaluateSafety';

const results = [
  evaluateIntent(),
  evaluateEntities(),
  evaluateReflection(),
  evaluateRecommendations(),
  evaluateRiskModels(),
  evaluateReports(),
  evaluateSafety(),
];

console.log('AltasAI internal model evaluation');
for (const result of results) {
  const status = result.passed ? 'PASS' : 'FAIL';
  console.log(`${status} ${result.name}: score=${result.score} threshold=${result.threshold}`);
}

console.log(JSON.stringify(results, null, 2));

if (results.some((result) => !result.passed)) {
  process.exitCode = 1;
}

File: backend/ml-service/app/evaluation/evaluation_runner.py
from app.evaluation.evaluate_entities import evaluate as evaluate_entities
from app.evaluation.evaluate_intent import evaluate as evaluate_intent
from app.evaluation.evaluate_rag import evaluate as evaluate_rag
from app.evaluation.evaluate_recommendation import evaluate as evaluate_recommendation
from app.evaluation.evaluate_risk import evaluate as evaluate_risk
from app.evaluation.evaluate_safety import evaluate as evaluate_safety
from app.evaluation.evaluate_vision import evaluate as evaluate_vision


def run_evaluation() -> dict:
    results = [
        evaluate_intent(),
        evaluate_entities(),
        evaluate_risk(),
        evaluate_recommendation(),
        evaluate_rag(),
        evaluate_safety(),
        evaluate_vision(),
    ]
    return {
        "passed": all(result["passed"] for result in results),
        "results": results,
    }


if __name__ == "__main__":
    result = run_evaluation()
    for item in result["results"]:
        status = "PASS" if item["passed"] else "FAIL"
        print(f"{status} {item['name']}: score={item['score']} threshold={item['threshold']}")
    if not result["passed"]:
        raise SystemExit(1)
